### ISSUE TITLE
problem with hyperparameters - TypeError

### ISSUE THREAD

**Issue by e-dorigatti** [marrlab/DomainLab#438]
```
processing dataset for diva
processing dataset for diva
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torch/nn/_reduction.py:42: UserWarning: size_average and reduce args will be deprecated, please use reduction='sum' instead.
  warnings.warn(warning.format(ret))
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torch/nn/init.py:412: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")

using device: cuda:0




!!!: not committed yet 
/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/utils/get_git_tag.py:17: UserWarning: !!!: not committed yet 
  warnings.warn("!!!: not committed yet")



b'44fee66'
model name: pathlist_te_sketch_diva_b44fee66_not_commited_2023md_09md_22_19_37_32_seed_0_221222
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: Metric `AUROC` will save all targets and predictions in buffer. For large datasets this may lead to large
  warnings.warn(*args, **kwargs)

 Experiment start at: 2023-09-22 19:37:33.168246
before training, model accuracy: 0.19572409987449646
[Fri Sep 22 19:37:46 2023]
Error in rule run_experiment:
    jobid: 0
    input: zoutput/benchmarks/benchmark_fbopt_pacs_full/hyperparameters.csv
    output: zoutput/benchmarks/benchmark_fbopt_pacs_full/rule_results/198.csv

RuleException:
TypeError in file /dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/benchmark.smk, line 151:
bad operand type for unary -: 'NoneType'
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/benchmark.smk", line 151, in __rule_run_experiment
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/run_experiment.py", line 131, in run_experiment
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/compos/exp/exp_main.py", line 62, in execute
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/algos/trainers/train_fbopt.py", line 59, in before_tr
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/algos/trainers/train_fbopt.py", line 113, in eval_r_loss
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/models/model_diva.py", line 137, in cal_reg_loss
  File "/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/concurrent/futures/thread.py", line 58, in run 
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
```processing dataset for diva
processing dataset for diva
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torch/nn/_reduction.py:42: UserWarning: size_average and reduce args will be deprecated, please use reduction='sum' instead.
  warnings.warn(warning.format(ret))
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torch/nn/init.py:412: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")

using device: cuda:0




!!!: not committed yet 
/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/utils/get_git_tag.py:17: UserWarning: !!!: not committed yet 
  warnings.warn("!!!: not committed yet")



b'44fee66'
model name: pathlist_te_sketch_diva_b44fee66_not_commited_2023md_09md_22_19_37_32_seed_0_221222
/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/site-packages/torchmetrics/utilities/prints.py:36: UserWarning: Metric `AUROC` will save all targets and predictions in buffer. For large datasets this may lead to large
  warnings.warn(*args, **kwargs)

 Experiment start at: 2023-09-22 19:37:33.168246
before training, model accuracy: 0.19572409987449646
[Fri Sep 22 19:37:46 2023]
Error in rule run_experiment:
    jobid: 0
    input: zoutput/benchmarks/benchmark_fbopt_pacs_full/hyperparameters.csv
    output: zoutput/benchmarks/benchmark_fbopt_pacs_full/rule_results/198.csv

RuleException:
TypeError in file /dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/benchmark.smk, line 151:
bad operand type for unary -: 'NoneType'
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/benchmark.smk", line 151, in __rule_run_experiment
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/exp_protocol/run_experiment.py", line 131, in run_experiment
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/compos/exp/exp_main.py", line 62, in execute
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/algos/trainers/train_fbopt.py", line 59, in before_tr
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/algos/trainers/train_fbopt.py", line 113, in eval_r_loss
  File "/dss/dssmcmlfs01/pr74ze/pr74ze-dss-0001/ru48zab2/projects/domainlab2/DomainLab/domainlab/models/model_diva.py", line 137, in cal_reg_loss
  File "/dss/dsshome1/lxc0A/ru48zab2/miniconda3/envs/domainlab/lib/python3.9/concurrent/futures/thread.py", line 58, in run 
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
```

**Comment by e-dorigatti** [comment]
config

```
198,diva_fbopt,diva,"{'ini_setpoint_ratio': 0.6000000000000001, 'k_i_gain': 0.08330000000000001, 'msel': 'last', 'msel_tr_loss': 'reg', 'es': 1}"
```

**Comment by smilesun** [comment]
this means hyperparameters for diva are NOne

**Comment by smilesun** [comment]
https://github.com/marrlab/DomainLab/blob/593c34499d96c6c84cfe5b4a809d800260bc5d41/domainlab/models/model_diva.py#L137 
this is the problem

**Comment by smilesun** [comment]
@e-dorigatti if we could find which multiplier is None that will solve half of the problem.
